ReactAndroid/src/main/java/com/facebook/react/modules/camera/CameraRollManager.java

```diff
@@ -386,11 +407,10 @@ private static boolean putImageInfo(
       WritableMap node,
       int idIndex,
       int widthIndex,
-      int heightIndex) {
+      int heightIndex,
+      int dataIndex) {
     WritableMap image = new WritableNativeMap();
-    Uri photoUri = Uri.withAppendedPath(
-        Images.Media.EXTERNAL_CONTENT_URI,
-        photos.getString(idIndex));
+    Uri photoUri = Uri.parse("file://" + photos.getString(dataIndex));
```

Comment:
Because it simply won't work to append the id directly to the external content uri.